experimental/dds/tree2/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -172,30 +178,27 @@ function composeMarks<TNodeChange>(
 		return withNodeChange(baseMark, nodeChange);
 	}
 	if (markIsTransient(baseMark)) {
-		if (isGenerativeMark(newMark)) {
+		if (isGenerate(newMark)) {
 			// TODO: Make `withNodeChange` preserve type information so we don't need to cast here
-			const nonTransient = withNodeChange(
-				baseMark,
-				nodeChange,
-			) as GenerativeMark<TNodeChange>;
-			delete nonTransient.transientDetach;
+			const nonTransient = withNodeChange(baseMark, nodeChange) as GenerateMark<TNodeChange>;
+			delete nonTransient.effects[0].transientDetach;
 			return nonTransient;
 		}
 		// Modify and Placeholder marks must be muted because the node they target has been deleted.
 		// Detach marks must be muted because the cell is empty.
-		if (newMark.type === "Modify" || newMark.type === "Placeholder" || isDetachMark(newMark)) {
+		if (isModify(newMark) || isPlaceholder(newMark) || isDetachMark(newMark)) {
```

Comment:
At this point this function always returns `baseMark`. I'm wondering whether all the cases and asserts are useful as documentation or if they are making things unecessarily complicated.

Reply:
I do think they are helpful in documenting our rationale. It would be nice if the code path for transients were not so custom. I'm hoping we might revisit that in the coming refactors.